src/interpreter/mod.rs

```diff
@@ -118,8 +118,8 @@ enum CachedMir<'mir, 'tcx: 'mir> {
 
 /// Represents the action to be taken in the main loop as a result of executing a terminator.
 enum TerminatorTarget {
-    /// Make a local jump to the given block.
-    Block(mir::BasicBlock),
+    /// Make a local jump to the next block
+    Block,
```

Comment:
Before it was that when going to a new block, it was the duty of the `eval_terminator` caller to actually change the block. But if `eval_terminator` did a function call, then `eval_terminator` updated the block (see here: https://github.com/solson/miri/pull/18/files/cc1ca73f5736a4aaa7a1ae164788051b8366fa3d..f995db9ffba925590e4c78917362998be02fcbc1#diff-b6ba58183c703f8ffcaa54879146b1c2R326)